hive-metastore/src/main/java/org/apache/iceberg/hive/HiveTableOperations.java

```diff
@@ -124,13 +184,37 @@ protected void doRefresh() {
       throw new RuntimeException("Interrupted during refresh", e);
     }
 
+    if (encryptionKeyIdFromHMS != null) {
+      encryptionKeyId = encryptionKeyIdFromHMS; // todo gg
+      encryptionDekLength =
+          (dekLengthFromHMS != null)
+              ? Integer.parseInt(dekLengthFromHMS)
+              : TableProperties.ENCRYPTION_DEK_LENGTH_DEFAULT; // todo gg
+    }
+
     refreshFromMetadataLocation(metadataLocation, metadataRefreshMaxRetries);
+
+    if (encryptionKeyIdFromHMS != null) {
+      checkEncryptionProperties(encryptionKeyIdFromHMS, dekLengthFromHMS);
+      encryptedKeysFromMetadata = current().encryptionKeys();
+    }
   }
 
   @SuppressWarnings({"checkstyle:CyclomaticComplexity", "MethodLength"})
   @Override
   protected void doCommit(TableMetadata base, TableMetadata metadata) {
     boolean newTable = base == null;
+    encryptionPropsFromMetadata(metadata.properties());
+
+    if (encryption() instanceof StandardEncryptionManager) {
+      TableMetadata.Builder builder = TableMetadata.buildFrom(metadata);
+      for (Map.Entry<String, EncryptedKey> entry :
+          EncryptionUtil.encryptionKeys(encryption()).entrySet()) {
```

Comment:
`EncryptionUtil.encryptionKeys` seems like a missing part of the PR, could you check please?

Reply:
yep, this PR is still not functional - it depends on #7770 , and also misses a few parts. Once the 7770 is merged, I'll update this one.